Movie page

AleksKors opened the pull request:
Movie page/info is almost done, but I haven't connected with the database yet. Also the tags above the description are not implemented yet. Not sure if it's still a feature?

AleksKors committed:
Merge branch 'main' of https://github.com/SE-E23-SP3/web-technology-project

AleksKors committed:
Merge remote-tracking branch 'origin/main' into movie-page

AleksKors committed:
First Commit

AleksKors committed:
Movie page

Can's load css, but should be finished

AleksKors committed:
Merge remote-tracking branch 'origin/main' into movie-page

AleksKors committed:
Fixed css

AleksKors committed:
Fixed ccs 2

AleksKors committed:
Small fix/addition

-Added button to rate
-Removed old artifacts

AleksKors committed:
Added rating  popup

AleksKors committed:
Merge branch 'main' of https://github.com/SE-E23-SP3/web-technology-project

AleksKors committed:
Somewhat connected DB with movie infopage

Missing rating interaction and a lot of the data displayed in figma is not saved in the DB?

MidnightRocket commented on routes/web.php:
```diff
@@ -39,3 +39,8 @@
 });
+Route::get('/movieinfo', function () {
+    return view('movies/movieinfo');
+})-> name('Movie Info');
```
Please rename named route, to something without space in it

MidnightRocket commented on resources/views/movies/movieinfo.blade.php:
```diff
@@ -0,0 +1,210 @@
+        <link rel="stylesheet" href="{{asset('css/movieinfo.css')}}">
+        <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
+        
+        <style>
```
Please move style into relevant style sheet

AleksKors committed:
Fix

Paknu22 committed:
Merge remote-tracking branch 'origin/main' into movie-page

AleksKors committed:
update to movieinfo

- Removed html from movieinfo that wasn't being used
- Added Cast & Crew section in movieinfo (Not connected to the new database yet)
- Changed route to "/movie/{movie_id}" instead of "/movieinfo/{movie_id}"
- Removed <style> section in movieinfo for the sake of Anders' sanity

AleksKors committed:
Merge remote-tracking branch 'origin/main' into movie-page

AleksKors committed:
Update to movieinfo

-MovieInfo now shows cast
-Renamed MovieInfo -> MovieInfoController
-Add to watchlist should be working now

AleksKors committed:
Second to last commit

-Added Rating
 -A user can only rate the same movie once
-The movie page now correctly displays the avg rating of the movie
-The movie page now correctly displays the amount of ratings
-Fixed the database seeder to seed the correct trailers

-----MISSING-----
The Crew displayed on the movie page currently only displays crew_type_id.
Idk how to accessm the type.. pls help Marco :(

Paknu22 committed:
Merge branch 'movie-page' of github.com:SE-E23-SP3/web-technology-project into movie-page

Paknu22 committed:
Character Fix

AleksKors committed:
Final push

-Now displays the correct crewtype insted of the ID

AleksKors committed:
Merge branch 'main' into movie-page

MidnightRocket commented on app/Http/Controllers/RateMovieController.php:
```diff
@@ -0,0 +1,27 @@
+{
+    $rating = $request->input('rating');
+    $user = auth()->user();
+    if ($user) {
```
Instead of doing unsafe type conversion use `if ($user != NULL)`

MidnightRocket commented on app/Http/Controllers/addToWatchlistController.php:
```diff
@@ -0,0 +1,24 @@
+public function addMovieToWatchlist(Request $request, Movie $movie) {
+    $user = auth()->user();
+
+    if ($user) {
```
Instead of doing unsafe type conversion use `if ($user != NULL)`

MidnightRocket commented on resources/views/movies/movieinfo.blade.php:
```diff
@@ -0,0 +1,221 @@
+            <article class="row">
+                <section class="col-3 offset-1">
+                    <article class="row">
+                        @isset($movie->title)
```
Instead of doing check if movie exists here. Do it in the controller, and return with `abort(Response::HTTP_NOT_FOUND)`, if movie is not found
Where must import  `use Illuminate\Http\Response;`
Should then remove all `isset` going forward

MidnightRocket commented on resources/views/movies/movieinfo.blade.php:
```diff
@@ -0,0 +1,221 @@
+                </section>
+                <section class="col-7">
+                    @isset($movie->trailers)
+                    @if($movie->trailers->first())
```
Do an `if($movie->trailers->first() != NULL)` instead of unsafe type casting with `if($movie->trailers->first())`

AleksKors committed:
Last push :)

-Removed Bootstrap

AleksKors committed:
Merge branch 'movie-page' of https://github.com/SE-E23-SP3/web-technology-project into movie-page

MidnightRocket commented on routes/web.php:
```diff
@@ -94,3 +97,15 @@
+
+Route::get('/movie/default', function () {
+    return view('movies/movieinfo');
+})->name('MovieInfo');
```
This route should properly be removed

MidnightRocket commented on routes/web.php:
```diff
@@ -94,3 +97,15 @@
+
+Route::post('/watchlist/add/{movie}', [addToWatchlistController::class, 'addMovieToWatchlist'])->name('watchlist.add')->middleware('auth');
+
+Route::post('/movies/{movie}/rate', [rateMovieController::class, 'rate'])->name('movies.rate');
```
Should use the auth middleware here.

MidnightRocket commented on app/Models/Movie.php:
```diff
@@ -34,6 +34,7 @@ public function addTrailer(string $string_url) {
     }
     public function addRating(User $user, int $rating) {
         if ($user instanceof User && is_int($rating)) {
+            $this->ratings()->detach($user->id);
```
Why first detach. And then attach again?

AleksKors replied:
In the case there the user already have rated the movie

AleksKors committed:
Fixed requested changes

MidnightRocket commented on app/Models/Movie.php:
```diff
@@ -34,6 +34,7 @@ public function addTrailer(string $string_url) {
     }
     public function addRating(User $user, int $rating) {
         if ($user instanceof User && is_int($rating)) {
```
Should use `if ($user != NULL && is_int($rating))` Instead, as the type constrain in the function argument list already forbids any other type.

MidnightRocket commented on app/Http/Controllers/RateMovieController.php:
```diff
@@ -0,0 +1,27 @@
+
+        return back()->with('message', 'Movie rated!');
+    } else {
+        return redirect()->back()->with('error', 'User not found!');
```
The should instead be `abort(Response::HTTP_NOT_FOUND, 'User not found');` instead, to better communicate the fatal error.

MidnightRocket commented on resources/views/movies/movieinfo.blade.php:
```diff
@@ -0,0 +1,205 @@
+        
+        <div class="container-fluid">
+            <article class="row">
+                <section class="col-3 offset-1">
+                    @isset($movie->poster_url)
+                        <img src="{{$movie->poster_url}}" class="pic">  
+                    @endisset
+                </section>
```
Maybe wrap isset around the section tag, so no empty section tag is generated if no poster url.

MidnightRocket commented on app/Http/Controllers/addToWatchlistController.php:
```diff
@@ -0,0 +1,24 @@
+        $user->addMovieToWatchlist($movie);
+        return redirect()->back()->with('message', 'Movie added to watchlist!');
+    } else {
+        return redirect()->back()->with('error', 'User not found!');
```
The should instead be abort(Response::HTTP_NOT_FOUND, 'User not found'); instead, to better communicate the fatal error.

AleksKors committed:
Fixed  requested changes

AleksKors committed:
Fixed requested changes

-Wrapped isset around section
